specs/market-api.yaml

```diff
@@ -827,27 +876,33 @@ components:
 
   requestBodies:
 
-
     Offer:
       required: true
       content:
         application/json:
           schema:
-            $ref: '#/components/schemas/Offer'
+            $ref: '#/components/schemas/DemandOfferBase'
 
     Demand:
       required: true
       content:
         application/json:
           schema:
-            $ref: '#/components/schemas/Demand'
+            $ref: '#/components/schemas/DemandOfferBase'
 
     Proposal:
       required: true
       content:
         application/json:
           schema:
-            $ref: '#/components/schemas/Proposal'
+            $ref: '#/components/schemas/DemandOfferBase'
+
+    Reason:
```

Comment:
Should reason be required?

Reply:
Here the `Reason` is a request body, which might or might not be required. 
Not much difference. It might be a better UX to allow an empty body also.

Aforementioned body has `Reason` schema which in turn has an optional `message` field.

Reply:
It's not good idea. Reason is optional, but message is required if reason exists. Otherwise we introduce another option which we will need to remove in future, like we want to remove optionals in proposals

Reply:
same here, look at my comment https://github.com/golemfactory/ya-client/pull/53#discussion_r516852151